### PR TITLE
Adapt to bitcoinj-thin parsers refactor

### PR DESCRIPTION
**Pull request by julia-zack**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java [hunk 1]
```diff
@@ -92,6 +94,12 @@ public static Sha256Hash createHash(int nHash) {
         return Sha256Hash.wrap(bytes);
     }
 
+    public static Script createBaseInputScriptThatSpendsFromTheRedeemScript(Script redeemScript) {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:97]: There a method here `createBaseRedeemScriptThatSpendsFromTheFederation` that seems like it's not used by anyone. And doesn't seem to make much sense either. Let's remove it

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java [hunk 2]
```diff
@@ -92,6 +94,12 @@ public static Sha256Hash createHash(int nHash) {
...
+        Script scriptPubKey = createP2SHOutputScript(redeemScript);
+        Script inputScript = scriptPubKey.createEmptyInputScript(null, redeemScript);
+        return inputScript;
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:100]:
```suggestion
        return scriptPubKey.createEmptyInputScript(null, redeemScript);
```
Simplify

File: rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java [hunk 3]
```diff
@@ -198,16 +195,24 @@ protected static boolean isValidPegInTx(
                 return false;
             }
 
+            TransactionInput txInput = tx.getInput(index);
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java:198]: `index` is declared above as `final int index = i;`. A bit weird, shouldn't we simply use `i`?

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java [hunk 4]
```diff
@@ -92,6 +94,12 @@ public static Sha256Hash createHash(int nHash) {
...
+    }
+
     public static Script createBaseInputScriptThatSpendsFromTheFederation(Federation federation) {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:103]:
```suggestion
    public static Script createBaseInputScriptThatSpendsFromTheFederation(Federation federation) {
         return createBaseInputScriptThatSpendsFromTheFederation(federation.getRedeemScript());
    }
```
That should do it

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsGetTransactionTypeTest.java [hunk 5]
```diff
@@ -1023,7 +1020,7 @@ void test_pegout_no_change_output(
     ) {
         // Arrange
         BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
-        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, activeFederation.getRedeemScript());
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, createBaseInputScriptThatSpendsFromTheFederation(activeFederation));
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsGetTransactionTypeTest.java:1023]:
```suggestion
        Script inputScript = createBaseInputScriptThatSpendsFromTheFederation(activeFederation);
        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, inputScript);
```
I'd split it like this to make lines shorter

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:97]: is being used in some tests, as there are some where we do not have the federation, but the redeem script

File: rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java:198]: the `scriptCorrectlySpendsTx` method takes a "final int", thats why the casting

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java (hunk 4 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:103]: you mean 
```
public static Script createBaseInputScriptThatSpendsFromTheFederation(Federation federation) {
         return createBaseInputScriptThatSpendsFromTheRedeemScript(federation.getRedeemScript());
    }
```

?

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java (hunk 1 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:97]: I mean another mehtod in this class `createBaseRedeemScriptThatSpendsFromTheFederation`. That one is not used anywhere

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java (hunk 4 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:102]: Yeap, my bad

File: rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java (hunk 3 above)
Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java:198]: oh ok, thanks

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsGetTransactionTypeTest.java (hunk 5 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsGetTransactionTypeTest.java]: Same in the other places where createBaseInputScriptThatSpendsFromTheFederation is called

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:97]: oh ok!

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**